### PR TITLE
Feat/ Terraform Module for Gcp Pub/Sub

### PR DESCRIPTION
**Pull request by kalyani-cd**
## what
* Improved the Google Cloud Pub/Sub Terraform module with enhancements for better topic management, refined IAM policy configuration

image

### CODE REVIEW

File: .github/workflows/tf-checks.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+  workflow_dispatch:
+jobs:
+  basic:
+    uses: clouddrove/github-shared-workflows/.github/workflows/tf-checks.yml@1.2.4
+    with:
+      working_directory: './examples/'
```
Comment (Tanveer143s) [inline .github/workflows/tf-checks.yml]: Add working directory proper for basic and complete both

File: .github/workflows/tfsec.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+permissions: write-all
+on:
+  pull_request:
+  workflow_dispatch:
+jobs:
+  tfsec:
+    uses: clouddrove/github-shared-workflows/.github/workflows/tfsec.yml@1.2.4
+    secrets: inherit
+    with:
+      working_directory: './examples/'
```
Comment (Tanveer143s) [inline .github/workflows/tfsec.yml]: update working directory

File: README.yaml [hunk 3]
```diff
@@ -5,40 +5,99 @@
...
 # How to use this project
 usage: |-
   Here are some examples of how you can use this module in your inventory structure:
   ```hcl
-    ```
+  module "pubsub" {
+  source  = "../"
+  
+  project_id = "clouddrove-1"
+  topic      = "topic-1"
```
Comment (Tanveer143s) [inline README.yaml:54]: Please fix tflint rules

File: .github/workflows/tf-checks.yml (hunk 1 above)
Comment (kalyani-cd) [reply .github/workflows/tf-checks.yml]: done sir

File: README.yaml (hunk 3 above)
Comment (kalyani-cd) [reply README.yaml:54]: done sir

File: .github/workflows/readme.yml [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+on:
+  push:
+    branches:
+      - master
+    paths-ignore:
+      - 'README.md'
+  workflow_dispatch:
+jobs:
+  assignee:
+    uses: clouddrove/github-shared-workflows/.github/workflows/readme.yml@master
```
Comment (VishwajitNagulkar) [inline .github/workflows/readme.yml]: use tag instaed of branch name

File: README.yaml [hunk 5]
```diff
@@ -5,40 +5,99 @@
 #
 
 # Name of this project
-name : Terraform Module Template
+name : Terraform Module for Google Cloud Pub/Sub
```
Comment (VishwajitNagulkar) [inline README.yaml]: use repo name as a project name `Terraform GCP Pubsub`

File: README.yaml [hunk 6]
```diff
@@ -5,40 +5,99 @@
...
+    image: "https://github.com/clouddrove/terraform-gcp-pubsub/actions/workflows/tfsec.yml/badge.svg"
+    url: "https://github.com/clouddrove/terraform-gcp-pubsub/actions/workflows/tfsec.yml"
+   - name: "Licence"
     image: "https://img.shields.io/badge/License-APACHE-blue.svg"
     url: "LICENSE.md"
+  - name: "Changelog"
+    image: "https://img.shields.io/badge/Changelog-blue"
+    url: "CHANGELOG.md"
 
 prerequesties:
```
Comment (VishwajitNagulkar) [inline README.yaml:31]: Only prerequesties where is `provider` field
plz refere following readme.yaml https://github.com/clouddrove/terraform-aws-eks/blob/master/README.yaml

File: .github/workflows/readme.yml (hunk 4 above)
Comment (kalyani-cd) [reply .github/workflows/readme.yml]: done sir

File: README.yaml (hunk 5 above)
Comment (kalyani-cd) [reply README.yaml]: done sir

File: README.yaml (hunk 6 above)
Comment (kalyani-cd) [reply README.yaml:31]: done sir

File: .github/workflows/readme.yml (hunk 4 above)
Comment (Tanveer143s) [reply .github/workflows/readme.yml]: use latest tag instead of 1.2.4

File: examples/basic/main.tf [hunk 7]
```diff
@@ -0,0 +1,38 @@
+
+# ------------------------------------------------------------------------------
+# Module
```
Comment (Tanveer143s) [inline examples/basic/main.tf]: add here identification of pull & push [like Module for push subscription and topic ]

File: examples/complete/main.tf [hunk 8]
```diff
@@ -0,0 +1,43 @@
+
+# ------------------------------------------------------------------------------
+# Module
```
Comment (Tanveer143s) [inline examples/complete/main.tf]: same add identification comment

File: .github/workflows/readme.yml (hunk 4 above)
Comment (kalyani-cd) [reply .github/workflows/readme.yml]: done sir

Comment (kalyani-cd) [reply .github/workflows/readme.yml]: done sir

File: examples/basic/main.tf (hunk 7 above)
Comment (kalyani-cd) [reply examples/basic/main.tf]: done sir

File: examples/complete/main.tf (hunk 8 above)
Comment (kalyani-cd) [reply examples/complete/main.tf]: done sir

**Review by Tanveer143s — APPROVED** [review]
LGTM!

**Review by VishwajitNagulkar — APPROVED** [review]

**Review by AshutoshAM2002 — APPROVED** [review]
LGTM

**Pull request merged**